runtime/compiler/ilgen/Walker.cpp

```diff
@@ -3092,13 +3092,63 @@ TR_J9ByteCodeIlGenerator::genInvokeInterface(int32_t cpIndex)
       genInstanceof(interfaceCPIndex);
       TR::Node *instanceof = pop();
 
+      // The receiver should not be NULL at this point because there should already be a NULLCHK
+      // before the call or there is no need of NULLCHK
+      instanceof->setReferenceIsNonNull(true);
+
       TR::SymbolReference *icce =
          symRefTab()->findOrCreateIncompatibleClassChangeErrorSymbolRef(_methodSymbol);
 
       TR::Node *check =
          TR::Node::createWithSymRef(TR::ZEROCHK, 1, 1, instanceof, icce);
 
-      callTree->insertBefore(TR::TreeTop::create(comp(), check));
+      TR::TreeTop *zeroCHKTT = callTree->insertBefore(TR::TreeTop::create(comp(), check));
+
+      /*
+       * If the class is unresolved, `genInstanceof` anchors `instanceof` under a `treetop` node.
+       * Then the anchored `instanceof` treetop appears after the callTree and it commons with
+       * the previous `instanceof` node under the ZEROCHK.
+       * This causes a problem because `expandUnresolvedClassInstanceof` does not expect the
+       * `treetop` that has the anchored `instanceof` to have already been evaluated when it appears
+       * under ZEROCHK. The transformation in `expandUnresolvedClassInstanceof` will not function correctly.
+       * Therefore, the anchored `instanceof` treetop needs to be moved up before ZEROCHK.
```

Comment:
I suggest removing the markdown from the inline comments.

Reply:
Addressed in 2f43cd6